eth/stagedsync/stage_headers_snapshot.go

```diff
@@ -24,77 +25,94 @@ func StageHeadersSnapshotGenCfg(db ethdb.RwKV, snapshotDir string) HeadersSnapsh
 }
 
 func SpawnHeadersSnapshotGenerationStage(s *StageState, tx ethdb.RwTx, cfg HeadersSnapshotGenCfg, sm *snapshotsync.SnapshotMigrator, torrentClient *snapshotsync.Client, quit <-chan struct{}) error {
-	useExternalTx := tx != nil
-	if !useExternalTx {
-		var err error
-		tx, err = cfg.db.BeginRw(context.Background())
-		if err != nil {
-			return err
-		}
-		defer tx.Rollback()
+	//generate snapshot only on initial mode
+	if tx != nil {
+		s.Done()
+		return nil
 	}
 
-	to, err := stages.GetStageProgress(tx, stages.Headers)
+	readTX, err := cfg.db.BeginRw(context.Background())
```

Comment:
If this is a `readTX`, why it is opened with `BeginRW` ? You should use `BeginRo` instead

Reply:
fixed